### DISCUSSION TITLE
FDC3 API not detected!

### DISCUSSION THREAD

**Discussion by ashishpatidar162** [finos/finos-pmcs#338] [General]
FDC3 API not detected!
An FDC3 desktop agent implementation was not found at window.fdc3.
I didn't found window.fdc3 function of file for changes.

**Comment by kriswest** [comment]
Did you try running your code in the FDC3 sandbox provided: https://citi-fdc3.cloud.interop.io/

**Reply by ashishpatidar162** [reply]
No. I am running on local setup

**Reply by kriswest** [reply]
A local setup including a Desktop Agent, such as FInsemble or io.Connect? Or just running in a browser? 

FDC3 is based on the concept of a Desktop Agent, which provides an API to the applications usually at `window.fdc3`. As that has to be injected Desktop Agents are usually based on technologies like Electron and replace a normal Web Browser. The sandbox we've provided is a Desktop Agent that runs in the browser, but currently requires a library to be used in your app to set up `window.fdc3` for you. See https://citi-fdc3.cloud.interop.io/ and click on the 'Learn how to use FDC3' link for instructions.

N.B. The next version of FDC3 will support running in a browser without adding a library to your ap through a feature known as FDC3 for  Web Browsers.

**Reply by ashishpatidar162** [reply]
Sure! Thanx @kriswest
